src/styles/blocks/calendar.scss

```diff
@@ -0,0 +1,51 @@
+.calendar {
+  position: relative;
+  @extend %flex-view--center;
+  @extend %font;
+  @extend %calendar-width;
+  margin: auto;
+  padding: $padding;
+  flex-wrap: wrap;
+
+  &__field {
+    display: flex;
+    flex-wrap: wrap;
+    gap: $gap-size;
+  }
+
+  &__day {
+    @extend %flex-view--center;
+    text-align: center;
+    width: $cell-size;
+    height: $cell-size;
+
+    background-color: #eee;
+    border: 1px solid #000;
+
+    @for $index from 1 through 31 {
+      &:nth-child(#{$index}) {
+        &::before {
+          content: "#{$index}";
+        }
+      }
+    }
+    &:hover {
+      cursor: pointer;
+      transform: translateY(-20px);
+      transition: 0.5s;
```

Comment:
add transition style under general selector, not the one with :hover - this way transition will work smoothly both ways.

Always add transition property. By default, transition property is all, and it's hard for the browser to control all properties.